src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -331,18 +521,436 @@ namespace Opm {
+        pind.push_back(ijk[1]*(dims[0]+1)*6 + ijk[0]*6);
+        pind.push_back(pind[0] + 6);
+        pind.push_back(pind[0] + (dims[0]+1)*6);
+        pind.push_back(pind[2] + 6);
+
+        // get depths from zcorn array in ZCORN array
+        zind.push_back(ijk[2]*dims[0]*dims[1]*8 + ijk[1]*dims[0]*4 + ijk[0]*2);
+        zind.push_back(zind[0] + 1);
+        zind.push_back(zind[0] + dims[0]*2);
+        zind.push_back(zind[2] + 1);
+
+        for (int n = 0; n < 4; n++) {
+            zind.push_back(zind[n] + dims[0]*dims[1]*4);
+        }
+
+        for (int n = 0; n< 8; n++) {
+            Z[n] = m_zcorn[zind[n]];
+        }
+
+        for (int  n=0; n<4; n++) {
+            double xt = m_coord[pind[n]];
+            double yt = m_coord[pind[n] + 1];
+            double zt = m_coord[pind[n] + 2];
+
+            double xb = m_coord[pind[n] + 3];
+            double yb = m_coord[pind[n] + 4];
+            double zb = m_coord[pind[n]+5];
+
+            X[n] = xt + (xb-xt) / (zt-zb) * (zt - Z[n]);
+            X[n+4] = xt + (xb-xt) / (zt-zb) * (zt-Z[n+4]);
+
+            Y[n] = yt+(yb-yt)/(zt-zb)*(zt-Z[n]);
+            Y[n+4] = yt+(yb-yt)/(zt-zb)*(zt-Z[n+4]);
+        }
+    }
+
+    std::vector<double> EclipseGrid::makeCoordDxvDyvDzvDepthz(const std::array<int, 3>& dims, const std::vector<double>& DXV, const std::vector<double>& DYV, const std::vector<double>& DZV, const std::vector<double>& DEPTHZ) {
+
+        std::vector<double> coord;
+        coord.reserve((dims[0]+1)*(dims[1]+1)*6);
+
+        for (int j = 0; j < dims[1] + 1; j++) {
+            for (int i = 0; i<dims[0] + 1; i++) {
+
+                double x0 = 0.0;
+
+                for (int n = 0; n < i; n++) {
+                    x0 = x0 + DXV[n];
+                }
+
+                double y0 = 0.0;
+
+                for (int n = 0; n < j; n++) {
+                    y0 = y0 + DYV[n];
+                }
+
+                size_t ind = i+j*(dims[0]+1);
+                double zb = DEPTHZ[ind];
+
+                for (int n = 0; n < dims[2]; n++) {
+                    zb = zb + DZV[n];
+                }
+
+                coord.push_back(x0);
+                coord.push_back(y0);
+                coord.push_back(DEPTHZ[ind]);
+                coord.push_back(x0);
+                coord.push_back(y0);
+                coord.push_back(zb);
+            }
+        }
+
+        return coord;
+    }
+
+    std::vector<double> EclipseGrid::makeZcornDzvDepthz(const std::array<int, 3>& dims, const std::vector<double>& DZV, const std::vector<double>& DEPTHZ) {
+
+        std::vector<double> zcorn;
+        size_t sizeZcorn = dims[0]*dims[1]*dims[2]*8;
+
+        zcorn.assign (sizeZcorn, 0.0);
+
+        for (int k = 0; k < dims[2]; k++) {
+            for (int j = 0; j < dims[1]; j++) {
+                for (int i = 0; i < dims[0]; i++) {
+
+                    double z = 0.0;
+
+                    for (int n = 0; n < k; n++) {
```

Comment:
As in the previous function, better to do a `partial_sum()` outside loop nest.

Reply:
done